subprojects/ivy/src/main/java/org/gradle/api/publish/ivy/internal/publication/DefaultIvyPublicationIdentity.java

```diff
@@ -64,4 +76,69 @@ public String getRevision() {
     public void setRevision(String revision) {
         this.revision = revision;
     }
+
+    private Module safeProjectCoordinatesProvider(Module module, Logger logger) {
+        if (module instanceof ProjectBackedModule) {
+            return new LazyProjectModuleProvider((ProjectBackedModule) module, logger);
+        }
+        return module;
+    }
+
+    private static final class LazyProjectModuleProvider implements Module {
+        private final ProjectBackedModule projectBackedModule;
+        private final Logger logger;
+        private final AtomicBoolean warned = new AtomicBoolean();
+
+        private LazyProjectModuleProvider(ProjectBackedModule module, Logger logger) {
+            this.projectBackedModule = module;
+            this.logger = logger;
+        }
+
+        @Nullable
+        @Override
+        public String getProjectPath() {
+            return projectBackedModule.getProjectPath();
+        }
+
+        @Override
+        public String getGroup() {
+            maybeWarnAboutDuplicates();
+            return projectBackedModule.getGroup();
+        }
+
+        @Override
+        public String getName() {
+            maybeWarnAboutDuplicates();
+            return projectBackedModule.getName();
+        }
+
+        @Override
+        public String getVersion() {
+            return projectBackedModule.getVersion();
+        }
+
+        @Override
+        public String getStatus() {
+            return projectBackedModule.getStatus();
+        }
+
+        private void maybeWarnAboutDuplicates() {
+            if (!warned.getAndSet(true)) {
+                Project project = projectBackedModule.getProject();
+                List<Project> projectsWithSameId = projectBackedModule.getProjectsWithSameCoordinates();
+                if (!projectsWithSameId.isEmpty()) {
+                    StringBuilder sb = new StringBuilder();
+                    sb.append("Project ")
+                        .append(project.getPath())
+                        .append(" has the same (organisation, module name) as ")
+                        .append(projectsWithSameId.stream().map(Project::getPath).collect(Collectors.joining(" and ")))
+                        .append(". It has been assigned an automatic module name of ")
```

Comment:
I wonder about the terminology: "automatic module name" has a very specific meaning in Java since 9.
Could we use the Maven terminology of `artifactId` or something else?

Reply:
So `module` is Ivy terminology. So why not use something like:
> It has been assigned a synthetic Ivy module name of

Wdyt?

Reply:
nice idea, done